The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9429013472/job/25974954881) [(plain)](https://github.com/rust-lang-ci/rust/commit/b79bc519ba8f1ddd629038ce8e9903aa013fe67b/checks/25974954881/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[TIMING] core::build_steps::tool::ToolBuild { compiler: Compiler { stage: 0, host: aarch64-apple-darwin }, target: aarch64-apple-darwin, tool: "tidy", path: "src/tools/tidy", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "" } -- 20.446
[TIMING] core::build_steps::tool::Tidy { compiler: Compiler { stage: 0, host: aarch64-apple-darwin }, target: aarch64-apple-darwin } -- 0.000
fmt check
fmt: skipped untracked file AWSCLIV2.pkg
##[error]Diff in /Users/runner/work/rust/rust/library/std/src/panicking.rs at line 768:
                 // recursive panics. However if the message is just a string, no user-defined
                 // code is involved in printing it, so that is risk-free.
                 let message: &str = payload.as_str().unwrap_or_default();
-                rtprintpanic!("panicked at {location}:\n{message}\nthread panicked while processing panic. aborting.\n");
+                rtprintpanic!(
+                    "panicked at {location}:\n{message}\nthread panicked while processing panic. aborting.\n"
             }
             panic_count::MustAbort::AlwaysAbort => {
                 // Unfortunately, this does not print a backtrace, because creating
                 // Unfortunately, this does not print a backtrace, because creating
fmt error: Running `"/Users/runner/work/rust/rust/build/aarch64-apple-darwin/rustfmt/bin/rustfmt" "--config-path" "/Users/runner/work/rust/rust" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/persist/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/persist/dirty_clean.rs" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/persist/save.rs" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/assert_dep_graph.rs" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/errors.rs" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/lib.rs" "/Users/runner/work/rust/rust/compiler/rustc_type_ir_macros/src/lib.rs" "/Users/runner/work/rust/rust/compiler/rustc_hir_pretty/src/lib.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/bufreader/buffer.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/mod.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/linewriter.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/linewritershim.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/bufreader.rs" "/Users/runner/work/rust/rust/library/std/src/io/buffered/bufwriter.rs" "/Users/runner/work/rust/rust/library/std/src/io/error/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/error/repr_bitpacked.rs" "/Users/runner/work/rust/rust/library/std/src/io/error/repr_unpacked.rs" "/Users/runner/work/rust/rust/library/std/src/io/prelude.rs" "/Users/runner/work/rust/rust/library/std/src/io/copy/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/stdio.rs" "/Users/runner/work/rust/rust/library/std/src/io/cursor.rs" "/Users/runner/work/rust/rust/library/std/src/io/stdio/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/impls.rs" "/Users/runner/work/rust/rust/library/std/src/io/mod.rs" "/Users/runner/work/rust/rust/library/std/src/io/impls/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/error.rs" "/Users/runner/work/rust/rust/library/std/src/io/cursor/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/util/tests.rs" "/Users/runner/work/rust/rust/library/std/src/io/util.rs" "/Users/runner/work/rust/rust/library/std/src/io/copy.rs" "/Users/runner/work/rust/rust/library/std/src/lib.rs" "/Users/runner/work/rust/rust/library/std/src/path/tests.rs" "/Users/runner/work/rust/rust/library/std/src/backtrace/tests.rs" "/Users/runner/work/rust/rust/library/std/src/env.rs" "/Users/runner/work/rust/rust/library/std/src/error.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/convert/mir.rs" "/Users/runner/work/rust/rust/library/std/src/hash/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/convert/abi.rs" "/Users/runner/work/rust/rust/library/std/src/hash/random.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/convert/ty.rs" "/Users/runner/work/rust/rust/library/std/src/panicking.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/convert/mod.rs" "/Users/runner/work/rust/rust/library/std/src/rt.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/convert/error.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/context.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/builder.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/alloc.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_smir/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/lib.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_internal/pretty.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_internal/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_smir/src/rustc_internal/internal.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/wtf8/tests.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/process.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/fs.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/tests.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/backtrace.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/mod.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/thread_local_dtor.rs" "/Users/runner/work/rust/rust/library/std/src/sys_common/lazy_box.rs" "/Users/runner/work/rust/rust/compiler/rustc/src/main.rs" "/Users/runner/work/rust/rust/compiler/rustc_incremental/src/persist/data.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Jun  8 13:15:13 UTC 2024
  network time: Sat, 08 Jun 2024 13:15:13 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/panic/panic_info.rs
+++ b/library/core/src/panic/panic_info.rs
@@ -24,14 +24,8 @@ pub struct PanicInfo<'a> {
 }
 
 impl<'a> PanicInfo<'a> {
-    #[unstable(
-        feature = "panic_internals",
-        reason = "internal details of the implementation of the `panic!` and related macros",
-        issue = "none"
-    )]
-    #[doc(hidden)]
     #[inline]
-    pub fn internal_constructor(
+    pub(crate) fn new(
         message: fmt::Arguments<'a>,
         location: &'a Location<'a>,
         can_unwind: bool,

--- a/library/core/src/panicking.rs
+++ b/library/core/src/panicking.rs
@@ -63,7 +63,7 @@ pub const fn panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
         fn panic_impl(pi: &PanicInfo<'_>) -> !;
     }
 
-    let pi = PanicInfo::internal_constructor(
+    let pi = PanicInfo::new(
         fmt,
         Location::caller(),
         /* can_unwind */ true,
@@ -101,12 +101,8 @@ pub const fn panic_nounwind_fmt(fmt: fmt::Arguments<'_>, force_no_backtrace: boo
         }
 
         // PanicInfo with the `can_unwind` flag set to false forces an abort.
-        let pi = PanicInfo::internal_constructor(
-            &fmt,
-            Location::caller(),
-            /* can_unwind */ false,
-            force_no_backtrace,
-        );
+        let pi =
+            PanicInfo::new(fmt, Location::caller(), /* can_unwind */ false, force_no_backtrace);
 
         // SAFETY: `panic_impl` is defined in safe Rust code and thus is safe to call.
         unsafe { panic_impl(&pi) }

--- a/library/std/src/panic.rs
+++ b/library/std/src/panic.rs
@@ -76,6 +76,7 @@ impl<'a> PanicInfo<'a> {
     /// panic!("Normal panic");
     /// ```
     #[must_use]
+    #[inline]
     #[stable(feature = "panic_hooks", since = "1.10.0")]
     pub fn payload(&self) -> &(dyn Any + Send) {
         self.payload
@@ -106,6 +107,7 @@ impl<'a> PanicInfo<'a> {
     /// panic!("Normal panic");
     /// ```
     #[must_use]
+    #[inline]
     #[stable(feature = "panic_hooks", since = "1.10.0")]
     pub fn location(&self) -> Option<&Location<'_>> {
         // NOTE: If this is changed to sometimes return None,
@@ -124,6 +126,7 @@ impl<'a> PanicInfo<'a> {
     /// false, however this will simply cause the panic handler to be called
     /// again.
     #[must_use]
+    #[inline]
     #[unstable(feature = "panic_can_unwind", issue = "92988")]
     pub fn can_unwind(&self) -> bool {
         self.can_unwind

--- a/library/std/src/panic.rs
+++ b/library/std/src/panic.rs
@@ -44,23 +44,14 @@ pub struct PanicInfo<'a> {
 }
 
 impl<'a> PanicInfo<'a> {
-    #[unstable(feature = "panic_internals", issue = "none")]
-    #[doc(hidden)]
     #[inline]
-    pub fn internal_constructor(
+    pub(crate) fn new(
         location: &'a Location<'a>,
+        payload: &'a (dyn Any + Send),
         can_unwind: bool,
         force_no_backtrace: bool,
     ) -> Self {
-        struct NoPayload;
-        PanicInfo { payload: &NoPayload, location, can_unwind, force_no_backtrace }
-    }
-
-    #[unstable(feature = "panic_internals", issue = "none")]
-    #[doc(hidden)]
-    #[inline]
-    pub fn set_payload(&mut self, info: &'a (dyn Any + Send)) {
-        self.payload = info;
+        PanicInfo { payload, location, can_unwind, force_no_backtrace }
     }
 
     /// Returns the payload associated with the panic.

--- a/library/std/src/panicking.rs
+++ b/library/std/src/panicking.rs
@@ -775,9 +775,7 @@ fn rust_panic_with_hook(
         crate::sys::abort_internal();
     }
 
-    let mut info = PanicInfo::internal_constructor(location, can_unwind, force_no_backtrace);
-    let hook = HOOK.read().unwrap_or_else(PoisonError::into_inner);
-    match *hook {
+    match *HOOK.read().unwrap_or_else(PoisonError::into_inner) {
         // Some platforms (like wasm) know that printing to stderr won't ever actually
         // print anything, and if that's the case we can skip the default
         // hook. Since string formatting happens lazily when calling `payload`
@@ -786,15 +784,12 @@ fn rust_panic_with_hook(
         // formatting.)
         Hook::Default if panic_output().is_none() => {}
         Hook::Default => {
-            info.set_payload(payload.get());
-            default_hook(&info);
+            default_hook(&PanicInfo::new(location, payload.get(), can_unwind, force_no_backtrace));
         }
         Hook::Custom(ref hook) => {
-            info.set_payload(payload.get());
-            hook(&info);
+            hook(&PanicInfo::new(location, payload.get(), can_unwind, force_no_backtrace));
         }
-    };
-    drop(hook);
+    }
 
     // Indicate that we have finished executing the panic hook. After this point
     // it is fine if there is a panic while executing destructors, as long as it

